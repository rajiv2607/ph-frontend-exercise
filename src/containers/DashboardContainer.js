import React from 'react'
import { connect } from 'react-redux'
import  LeftPanelContainer  from './LeftPanelContainer'
import RightPanelContainer from './RightPanelContainer'
import  './dashboard-container.css';
import navigation from '../assets/navigation.json'

export const DashboardContainer = (props) => {
    return (
        <div className='dashboard-container'>
            <LeftPanelContainer navigation={navigation}/>
            <RightPanelContainer/>
        </div>
    )
}

const mapStateToProps = (state) => ({
    
})

const mapDispatchToProps = {
    
}

export default connect(mapStateToProps, mapDispatchToProps)(DashboardContainer)
