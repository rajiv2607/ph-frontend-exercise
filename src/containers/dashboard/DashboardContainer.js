import React, {useState} from 'react'
import { connect } from 'react-redux'
import  LeftPanelContainer  from '../leftPanelContainer/LeftPanelContainer'
import RightPanelContainer from '../rightPanelContainer/RightPanelContainer'
import  './dashboard-container.css';
import navigation from '../../assets/navigation.json'
import {useFetch} from '../../hooks/useFetch'

 const DashboardContainer = (props) => {

    const URL = 'https://run.mocky.io/v3/b49604f2-3705-4e14-992f-1378fb4b598f?mocky-delay=1000ms'
    const  {response, error} = useFetch(URL);

    return (
        <div className='dashboard-container'>
            {
                response && <>
                    <LeftPanelContainer navigation={response}/>
                    <RightPanelContainer/>
                </>
            }
        </div>
    )
}

export default DashboardContainer;
