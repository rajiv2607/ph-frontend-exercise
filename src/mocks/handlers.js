import { rest } from 'msw';
import navigation from '../__mock__/navigation.json'
export const handlers = [
    rest.get('https://run.mocky.io/v3/', (req, res, ctx) => {
        return res(
            ctx.json(navigation)
        );
    }),
];
