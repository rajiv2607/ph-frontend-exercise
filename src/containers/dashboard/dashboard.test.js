import  '@testing-library/jest-dom';
import renderer from 'react-test-renderer';

// src/setupTests.js
import { server } from '../../mocks/server';
import DashboardContainer from "./DashboardContainer";
// Establish API mocking before all tests.
beforeAll(() => server.listen());

// Reset any request handlers that we may add during the tests,
// so they don't affect other tests.
afterEach(() => server.resetHandlers());

// Clean up after the tests are finished.
afterAll(() => server.close());


describe ('test using mock', () => {
    test('mocking api and snapshot test ' , () => {
        const component = renderer.create(
            <DashboardContainer/>
        );
        const tree = component.toJSON();
        expect(tree).toMatchSnapshot();
    })
})
