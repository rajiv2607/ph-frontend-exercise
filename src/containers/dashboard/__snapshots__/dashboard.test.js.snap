// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`test using mock mocking api and snapshot test  1`] = `
<div
  className="dashboard-container"
/>
`;
